### DISCUSSION TITLE
ESPAltherma on two systems on Home Assistant

### DISCUSSION THREAD

**Discussion by DelanoBruggink** [raomin/ESPAltherma#469] [Q&A]
Hello everyone

I have been running the ESPAltherma successfully, but I have two systems and I can't get them both to work at the same time.
I now have them both on a separate M5stickC plus2. When I try it on an ESP32 and an M5stickC plus2 it doesn't work either 

can anyone help me further.

**Comment by aliban** [comment]
Did you change the topic definition on the second unit. Edit setup.php and in the second unit change  #define MQTT_OneTopic "emon/espaltherma/OneATTR/"  to  #define MQTT_OneTopic "emon/espaltherma2/OneATTR/"

**Reply by DelanoBruggink** [reply]
I tried your tip but it doesn't work.

system 1:
//Uncomment this if you want to activate the One Value <-> One Topic mode. Each value will be sent to a specific topic below 
#define ONEVAL_ONETOPIC
#define MQTT_OneTopic "espaltherma1/OneATTR/" //Keep the ending "/" !!

System 2:
//Uncomment this if you want to activate the One Value <-> One Topic mode. Each value will be sent to a specific topic below 
#define ONEVAL_ONETOPIC
#define MQTT_OneTopic "espaltherma2/OneATTR/" //Keep the ending "/" !!

the systems still alternate between the two

**Reply by aliban** [reply]
Can you share the view of MQTT explorer, now you should have two topics.

**Reply by DelanoBruggink** [reply]
I got it working.
in the folder include/mqtt.h I changed all espaltherma references to espaltherma1 /espaltherma2.

It might not be the right way to do it but it works

**Reply by aliban** [reply]
Yes, sorry about that, but this will be a nice workarounf
